plugin/web/extensions/sidebar/utils/dataset.ts

```diff
@@ -49,3 +51,11 @@ export const getDefaultGroup = (components?: FieldComponent[], baseTemplates?: T
     return switchGroupComponents[switchGroupComponents.length - 1].groups[0].fieldGroupID;
   }
 };
+
+export const getDefaultDataset = (dataset?: DataCatalogItem | UserDataItem) => {
+  if (!dataset) return;
+
+  if (dataset.config?.data && dataset.config?.data?.length > 0) {
+    return dataset.config.data[0];
+  }
```

Comment:
I believe this needs to fallback to dataset.url. Please check the createLayer function, @airslice recently updated how we set the dataset's data.type (aka selected dataset url) there so should be synced (even better would be to use dataset.selectedDataset in the createLayer as well)